cmd/agent/app/reporter/grpc/builder.go

```diff
@@ -82,18 +82,26 @@ func (b *ConnBuilder) CreateConnection(logger *zap.Logger, mFactory metrics.Fact
 		if b.CollectorHostPorts == nil {
 			return nil, errors.New("at least one collector hostPort address is required when resolver is not available")
 		}
-		if len(b.CollectorHostPorts) > 1 {
+		var collectorEndpoints []string
```

Comment:
```suggestion
		collectorEndpoints := fixHosts(b.CollectorHostPorts)
```
this way we can add a unit test for `fixHosts` function to keep the coverage up